File: webversion/server/test_signaling.py
import asyncio
import websockets
import json

async def web_client():
    async with websockets.connect("ws://localhost:3000") as ws:
        welcome = await ws.recv()
        print(f"[Web] Recv: {welcome}")
        
        await ws.send(json.dumps({
            "type": "join",
            "room": "teleport-default",
            "name": "Web Client",
            "clientType": "web"
        }))
        
        peers = await ws.recv()
        print(f"[Web] Recv: {peers}")
        
        while True:
            msg = await ws.recv()
            print(f"[Web] Recv: {msg}")

async def desktop_client():
    await asyncio.sleep(2) # Connect after Web
    async with websockets.connect("ws://localhost:3000") as ws:
        welcome = await ws.recv()
        print(f"[Desktop] Recv: {welcome}")
        
        # Desktop sends exactly this string format:
        payload = "{\"type\":\"join\",\"room\":\"teleport-default\",\"name\":\"Desktop Client\",\"platform\":\"desktop\"}"
        await ws.send(payload)
        
        peers = await ws.recv()
        print(f"[Desktop] Recv: {peers}")
        
        while True:
            msg = await ws.recv()
            print(f"[Desktop] Recv: {msg}")

async def main():
    await asyncio.gather(
        web_client(),
        desktop_client()
    )

asyncio.run(main())

File: webversion/server/test_signaling.js
const WebSocket = require('ws');

function runWebClient() {
    const ws = new WebSocket('ws://localhost:3000');
    ws.on('message', data => {
        console.log('[Web] Recv:', data.toString());
        const msg = JSON.parse(data);
        if (msg.type === 'welcome') {
            ws.send(JSON.stringify({
                type: 'join',
                room: 'teleport-default',
                name: 'Web Client',
                clientType: 'web'
            }));
        }
    });
}

function runDesktopClient() {
    const ws = new WebSocket('ws://localhost:3000');
    ws.on('message', data => {
        console.log('[Desktop] Recv:', data.toString());
        const msg = JSON.parse(data);
        if (msg.type === 'welcome') {
            ws.send("{\"type\":\"join\",\"room\":\"teleport-default\",\"name\":\"Desktop Client\",\"platform\":\"desktop\"}");
        }
    });
}

runWebClient();
setTimeout(runDesktopClient, 2000);

setTimeout(() => process.exit(0), 5000);
